packages/immutable-rootfs/30cos-immutable-rootfs/module-setup.sh

```diff
@@ -20,13 +20,24 @@ installkernel() {
 install() {
     declare moddir=${moddir}
     declare systemdutildir=${systemdutildir}
+    declare systemdsystemunitdir=${systemdsystemunitdir}
+    declare initdir="${initdir}"
+
     inst_multiple \
-        mount mountpoint yip cos-setup sort rmdir
-    inst_hook cmdline 30 "${moddir}/parse-cos-overlay.sh"
-    inst_hook initqueue/finished 30 "${moddir}/cos-wait-oem.sh"
-    inst_hook pre-pivot 10 "${moddir}/cos-mount-layout.sh"
-    inst_hook pre-pivot 20 "${moddir}/cos-config-launcher.sh"
+        mount mountpoint yip cos-setup sort findmnt rmdir
+    inst_hook cmdline 30 "${moddir}/parse-cos-cmdline.sh"
     inst_script "${moddir}/cos-generator.sh" \
         "${systemdutildir}/system-generators/dracut-cos-generator"
+    inst_script "${moddir}/cos-mount-layout.sh" "/sbin/cos-mount-layout"
+    inst_script "${moddir}/cos-loop-img.sh" "/sbin/cos-loop-img"
+    inst_simple "${moddir}/cos-immutable-rootfs.service" \
+        "${systemdsystemunitdir}/cos-immutable-rootfs.service"
+    inst_simple "${moddir}/cos-setup-rootfs.service" \
+        "${systemdsystemunitdir}/cos-setup-rootfs.service"
+    mkdir -p "${initdir}/${systemdsystemunitdir}/initrd-fs.target.requires"
+    ln_r "../cos-immutable-rootfs.service" \
+        "${systemdsystemunitdir}/initrd-fs.target.requires/cos-immutable-rootfs.service"
+    ln_r "../cos-setup-rootfs.service" \
```

Comment:
I added the work in progress mostly because I think this service should be part of `cos-setup` package, I just did not have time enough to do the change when I worked on that two weeks ago.